File: be/pnt_spring/src/main/java/com/pnt/pnt_spring/domain/auth/api/resp/LoginResponse.java
package com.pnt.pnt_spring.domain.auth.api.resp;

import com.pnt.pnt_spring.domain.auth.api.req.TokenDto;
import com.pnt.pnt_spring.domain.members.member.entity.Member;
import com.pnt.pnt_spring.domain.members.member.entity.MemberProfile;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class LoginResponse {

    private String accessToken;
    private String refreshToken;
    private LoginMemberInfo member;

    @Getter
    @Builder
    public static class LoginMemberInfo {
        private String id; // 로그인 아이디
        private String nickname;
        private String avatarUrl;
        private String role;
    }

    // 응답 객체 생성
    public static LoginResponse of(TokenDto tokenDto, Member member, MemberProfile profile) {
        return LoginResponse.builder()
                .accessToken(tokenDto.getAccessToken())
                .refreshToken(tokenDto.getRefreshToken())
                .member(LoginMemberInfo.builder()
                        .id(member.getLoginId())
                        .nickname(profile.getNickname())
                        .avatarUrl(profile.getAvatarUrl())
                        .role("USER")
                        .build())
                .build();
    }
}

File: fe/D104/app/src/main/java/com/d104/pnt/data/remote/model/response/LoginResponse.kt
package com.d104.pnt.data.remote.model.response

import com.google.gson.annotations.SerializedName

/**
 * 로그인 응답
 */
data class LoginResponse(
    @SerializedName("accessToken")
    val accessToken: String,

    @SerializedName("refreshToken")
    val refreshToken: String,

    @SerializedName("member")
    val member: MemberData
)

/**
 * 회원 정보
 */
data class MemberData(
    @SerializedName("memberId")
    val memberId: Long,

    @SerializedName("id")
    val id: String,

    @SerializedName("nickname")
    val nickname: String?,  // nullable

    @SerializedName("avatarUrl")
    val avatarUrl: String?,

    @SerializedName("role")
    val role: String?  // nullable
)